xarray/backends/api.py

```diff
@@ -1358,7 +1370,9 @@ def to_netcdf(
 
         writes = writer.sync(compute=compute)
 
-        if isinstance(target, BytesIO):
+        if engine == "netcdf4" and path_or_file is None:
+            return store._manager.close().tobytes()
```

Comment:
The `finally` statement is called before returning, I am not sure I which order.
Might that (=`store.close()`) cause issues?
Maybe we could add a test for that...